relentless/potential/potential.py

```diff
@@ -1,29 +1,118 @@
+    """ Pair coefficient matrix.
+
+    Defines a matrix of PairMatrix objects with one or more parameters.
+    Any or all of the parameters can be initialized with a default value.
+
+    Parameters
+    ----------
+    types : array_like
+        List of types (A type must be a `str`).
+    params : array_like
+        List of parameters (A parameter must be a `str`).
+    default : float or int
+        Initial value for any or all parameters, defaults to `None`.
+
+    Raises
+    ------
+    TypeError
+        If params does not consist of only strings
+
+    Examples
+    --------
+    Create a coefficient matrix::
+
+        m = CoefficientMatrix(types=('A','B'), params=('energy','mass'))
+
+    Create a coefficient matrix with a default value::
+
+        m = CoefficientMatrix(types=('A','B'), params=('energy','mass'), default=0.0)
+
+    Set coefficient matrix values by accessing parameter directly::
+
+        m['A','A']['energy'] = 2.0
+
+    Assigning to a pair using `update()` overwrites parameters::
+
+        m['A','A'].update({'mass':2.5})  #does not reset 'energy' value to default
+        m['A','A'].update(mass=2.5)      #equivalent statement
+        >>> print(m['A','A'])
+        {'energy':2.0, 'mass':2.5}
+
+    Assigning to a pair using `=` operator overwrites parameters and resets any default::
+
+        m['A','A'] = {'mass':2.5}  #does reset 'energy' value to default
+        >>> print(m['A','A'])
+        {'energy':0.0, 'mass':2.5}
+
+    Set coefficient matrix values by setting parameters in full::
+
+        m['A','B'] = {'energy':Variable(value=2.0,high=1.5), 'mass':0.5}
+
+    Set coefficient matrix values by iteratively accessing parameters::
+
+        for p in m.params:
+            m['B','B'][p] = 0.1
+
+    Evaluate (retrieve) pair parameters::
+
+        >>> print(m.evaluate(('B','B')))
+        {'energy':0.1, 'mass':0.1}
+
+    Utilizing `evaluate()` computes the defined values of all objects,
+    while directly accessing values returns the objects themselves::
+
+        >>> print(m.evaluate(('A','B')))
+        {'energy':2.0, 'mass':0.5}
+        >>> print(m['A','B'])
+        {'energy':<relentless.core.Variable object at 0x561124456>, 'mass':0.5}
+
+    """
+    def __init__(self, types, params, default=None):
         super().__init__(types)
+        if not all(isinstance(p, str) for p in params):
+            raise TypeError('All parameters must be strings')
         self.params = tuple(params)
+        self.default = default
         for key in self:
-            for p in self.params:
-                v = default[p] if p in default else None
-                self[key][p] = v
+            self[key] = FixedKeyDict(keys=self.params,default=self.default)
 
     def evaluate(self, pair):
         """Evaluate pair parameters.
 
+        Returns a dictionary of the parameters and values for the specified pair.
+
+        Parameters
+        ----------
+        pair : array_like
```

Comment:
Can `pair` be array_like, or does it need to be a tuple? I forget how the indexing is defined.